src/components/AddBook.js

```diff
@@ -1,12 +1,39 @@
 import React from 'react';
+import { v4 as uuidv4 } from 'uuid';
+import { useDispatch } from 'react-redux';
+import { addBook } from '../redux/books/books';
 
-const AddBook = () => (
-  <form className="newbook">
-    <h2>ADD NEW BOOK</h2>
-    <input type="text" placeholder="Book title" />
-    <input type="text" placeholder="Book author" />
-    <button type="submit">Add Book</button>
-  </form>
-);
+const AddBook = () => {
+  const dispatch = useDispatch();
+
+  const actionSubmit = (e) => {
+    e.preventDefault();
+    const newTitle = e.target.title.value;
+    const newAuthor = e.target.author.value;
+
+    const newBook = {
+      id: uuidv4(),
+      title: newTitle,
+      author: newAuthor,
+    };
+
+    if (newTitle.trim() && newAuthor.trim()) {
+      dispatch(addBook(newBook));
+      e.target.title.value = '';
+      e.target.author.value = '';
+    } else {
+      alert('Please write item');
+    }
```

Comment:
- [ ] Please kindly check your implementation on the add functionality as it is currently not adding the book to the store.

Hint:
- Put the `OnSubmit` in the `form` instead of the `button`
- Also, create states to hold the input value and use the `onChange` event to update the state.